src/rules/no-missing-link-fragments.js

```diff
@@ -57,81 +78,101 @@ export default {
 		],
 
 		messages: {
-			missingFragment:
-				"Link fragment '#{{fragment}}' does not exist in the document.",
+			invalidFragment:
+				"Link fragment '{{fragment}}' does not reference a valid heading or anchor.",
 		},
 	},
 
 	create(context) {
 		const options = context.options[0] || {};
-		const ignoreCase = options.ignoreCase;
-		const allowPattern = options.allowPattern;
-
-		const allowedRegex = allowPattern
-			? new RegExp(allowPattern, "u")
+		const ignoreCase = options.ignoreCase || false;
+		const allowPattern = options.allowPattern
+			? new RegExp(options.allowPattern, "u")
 			: null;
 
-		const headingIds = new Set();
-		const lowercaseHeadingIds = new Set();
+		const slugger = new GithubSlugger();
```

Comment:
Let's move this into the global scope. There's no need to create a new reference each time the rule is created.